tests/evaluate/test_evaluation.py

```diff
@@ -2139,3 +2140,26 @@ def test_import_evaluation_dataset():
     # This test is to validate both imports work at the same time
     from mlflow.models.evaluation import EvaluationDataset
     from mlflow.models.evaluation.base import EvaluationDataset  # noqa: F401
+
+
+def test_evaluate_shows_server_stdout_and_stderr_on_error(linear_regressor_model_uri, diabetes_dataset):
+    with mlflow.start_run():
+        server_proc = subprocess.Popen("foo", shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
why `shell=True`?

Reply:
```suggestion
        server_proc = subprocess.Popen(["foo"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Reply:
if you try `subprocess.Popen(["foo"])`, the Popen invocation directly raises error without creating child process

Reply:
I see, then you can use `["echo", "foo"]`

Reply:
but `echo foo` command succeeds. we need a command to return non-zero value

Reply:
Does that matter in this test?

Reply:
updated!